src/ol/source/imagestaticsource.js

```diff
@@ -26,10 +26,15 @@ ol.source.ImageStatic = function(options) {
 
   var imageExtent = options.imageExtent;
 
-  var resolution, resolutions;
+  var xResolution, yResolution, resolutions;
   if (options.imageSize !== undefined) {
-    resolution = ol.extent.getHeight(imageExtent) / options.imageSize[1];
-    resolutions = [resolution];
+    xResolution = ol.extent.getWidth(imageExtent) / options.imageSize[0];
+    yResolution = ol.extent.getHeight(imageExtent) / options.imageSize[1];
+    if (xResolution !== yResolution) {
+      resolutions = [Math.min(xResolution, yResolution), Math.max(xResolution, yResolution)];
```

Comment:
It was in the original patch and I didn't really think it through, let me see if this is actually needed, probably not indeed.